### PR TITLE
CMakeLists.txt: do not compile {Objecter,Striper}.cc twice

### PR DESCRIPTION
**Pull request by MaxKellermann**
Was part of https://github.com/ceph/ceph/pull/60175 (split upon @batrick's request)

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [X] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [X] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [X] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [X] No tests

### CODE REVIEW

File: src/osdc/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,7 @@
 set(osdc_files
   Filer.cc
   ObjectCacher.cc
-  Objecter.cc
-  error_code.cc
-  Striper.cc)
+  error_code.cc)
```
Comment (batrick) [inline src/osdc/CMakeLists.txt:4]: Please add a comment. That's pretty confusing.

Comment (MaxKellermann) [reply src/osdc/CMakeLists.txt:4]: I don't understand - what confuses you? That some source files in this directory are missing? Others are missing, too, but there is no comment.

Comment (batrick) [reply src/osdc/CMakeLists.txt:4]: > That some source files in this directory are missing?

Yes

> Others are missing, too, but there is no comment.

Extra confusing!

Comment (MaxKellermann) [reply src/osdc/CMakeLists.txt:4]: OK, I added a comment

**Review by batrick — APPROVED** [review]

**Comment by MaxKellermann** [comment]
@batrick, how's QA going?

**Comment by cbodley** [comment]
> @batrick, how's QA going?

this pr is labeled `needs-qa` but not labeled for any ceph component so nobody is likely to pick it up for testing

i've pushed this branch to ceph-ci to check for build regressions: https://shaman.ceph.com/builds/ceph/pr-60217/

@batrick are you sure this needs qa?

**Comment by batrick** [comment]
> > @batrick, how's QA going?
> 
> this pr is labeled `needs-qa` but not labeled for any ceph component so nobody is likely to pick it up for testing
> 
> i've pushed this branch to ceph-ci to check for build regressions: https://shaman.ceph.com/builds/ceph/pr-60217/
> 
> @batrick are you sure this needs qa?

Mostly I thought it should simply be built with a test branch to confirm nothing unexpected happens.

**Comment by batrick** [comment]
Adding core since `osdc` is under their purview (mostly).

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
jenkins test api

**Comment by cbodley** [comment]
> i've pushed this branch to ceph-ci to check for build regressions: https://shaman.ceph.com/builds/ceph/pr-60217/

builds green in https://shaman.ceph.com/builds/ceph/pr-60217/c1e5d21430dde72c032ab618b53382a6ea068062/ (had to rebase the branch)

**Comment by batrick** [comment]
jenkins test api

**Comment by Naveenaidu** [comment]
Rados approved: https://tracker.ceph.com/issues/69215#note-2

**Pull request merged**